metadata-ingestion/src/datahub/ingestion/source/sac/sac.py

```diff
@@ -0,0 +1,808 @@
+                        },
+                        datasets=sorted(datasets) if datasets else None,
+                        externalUrl=f"{self.config.tenant_url}{resource.open_url}",
+                    ),
+                )
+
+                yield MetadataWorkUnit(id=f"dashboard-info-{dashboard_urn}", mcp=mcp)
+
+                type_name: Optional[str] = None
+                if resource.resource_subtype == "":
+                    type_name = BIAssetSubTypes.SAC_STORY
+                elif resource.resource_subtype == "APPLICATION":
+                    type_name = BIAssetSubTypes.SAC_APPLICATION
+
+                if type_name:
+                    mcp = MetadataChangeProposalWrapper(
+                        entityType="dashboard",
+                        changeType=ChangeTypeClass.UPSERT,
+                        entityUrn=dashboard_urn,
+                        aspectName="subTypes",
+                        aspect=SubTypesClass(
+                            typeNames=[type_name],
+                        ),
+                    )
+
+                    yield MetadataWorkUnit(
+                        id=f"dashboard-subtype-{dashboard_urn}", mcp=mcp
+                    )
+
+    def get_report(self) -> SACSourceReport:
+        return self.report
+
+    def get_model_workunits(
+        self, dataset_urn: str, model: ResourceModel
+    ) -> Iterable[MetadataWorkUnit]:
+        mcp = MetadataChangeProposalWrapper(
+            entityType="dataset",
+            changeType=ChangeTypeClass.UPSERT,
+            entityUrn=dataset_urn,
+            aspectName="datasetProperties",
+            aspect=DatasetPropertiesClass(
+                name=model.name,
+                description=model.description,
+                customProperties={
+                    "namespace": model.namespace,
+                    "modelId": model.model_id,
+                    "isImport": "true" if model.is_import else "false",
+                },
+                externalUrl=f"{self.config.tenant_url}/sap/fpa/ui/tenants/3c44c#view_id=model;model_id={model.namespace}:{model.model_id}",
+            ),
+        )
+
+        yield MetadataWorkUnit(id=f"dataset-properties-{dataset_urn}", mcp=mcp)
+
+        if model.is_import:
+            primary_fields: List[str] = []
+            schema_fields: List[SchemaFieldClass] = []
+
+            columns = self.get_import_data_model_columns(model_id=model.model_id)
+            for column in columns:
+                native_data_type = column.data_type
+                if column.data_type == "decimal":
+                    native_data_type = (
+                        f"{column.data_type}({column.precision}, {column.scale})"
+                    )
+                elif column.data_type == "int32":
+                    native_data_type = f"{column.data_type}({column.precision})"
+                elif column.max_length is not None:
+                    native_data_type = f"{column.data_type}({column.max_length})"
```

Comment:
would be nicer to move this into a helper method called `get_schema_field_native_data_type`